src/sentry/lang/native/plugin.py

```diff
@@ -238,41 +248,45 @@ def process_frame(self, processable_frame, processing_task):
                 # optional dsyms)
                 errors = []
                 if e.is_user_fixable or e.is_sdk_failure:
-                    errors.append(
-                        {
-                            'type': e.type,
-                            'image_uuid': e.image_uuid,
-                            'image_path': e.image_path,
-                            'image_arch': e.image_arch,
-                            'message': e.message,
-                        }
-                    )
+                    errors.append({
+                        'type': e.type,
+                        'image_uuid': e.image_uuid,
+                        'image_path': e.image_path,
+                        'image_arch': e.image_arch,
+                        'message': e.message,
+                    })
                 else:
-                    logger.debug('Failed to symbolicate with native backend', exc_info=True)
+                    logger.debug('Failed to symbolicate with native backend',
+                                 exc_info=True)
+
                 return [raw_frame], [raw_frame], errors
 
             processable_frame.set_cache_value([in_app, symbolicated_frames])
-        else:
+
+        else:  # processable_frame.cache_value is present
             in_app, symbolicated_frames = processable_frame.cache_value
             raw_frame['in_app'] = in_app
 
+        new_frames = []
         for sfrm in symbolicated_frames:
             new_frame = dict(frame)
             new_frame['function'] = sfrm['function']
             if sfrm.get('symbol'):
                 new_frame['symbol'] = sfrm['symbol']
             new_frame['abs_path'] = sfrm['abs_path']
             new_frame['filename'] = sfrm.get('filename') or \
-                (sfrm['abs_path'] and posixpath.basename(sfrm['abs_path'])) or None
+                (sfrm['abs_path'] and posixpath.basename(sfrm['abs_path'])) or \
+                None
             if sfrm.get('lineno'):
                 new_frame['lineno'] = sfrm['lineno']
             if sfrm.get('colno'):
                 new_frame['colno'] = sfrm['colno']
-            if sfrm.get('package'):
-                new_frame['package'] = sfrm['package']
+            if sfrm.get('package') or processable_frame.data['obj'] is not None:
+                new_frame['package'] = sfrm.get(
+                    'package', processable_frame.data['obj'].name)
```

Comment:
We have to use the object's name here, as the `package` field is missing in events generated from minidumps. While the information would be present in Breakpad, we stopped to expose it in symbolic. In general, I also like the idea of reading it from the object more than having it in every frame.